packages/core/src/workflows/workflow.ts

```diff
@@ -489,11 +490,13 @@ export class Workflow<
         return await otlpContext.with(
           trace.setSpan(otlpContext.active(), this.getExecutionSpan(attributes?.runId ?? data?.runId) as Span),
           async () => {
-            // @ts-ignore
-            return this.#mastra.telemetry.traceMethod(handler, {
-              spanName,
-              attributes,
-            })(data);
+            if (this?.telemetry) {
+              return this.telemetry.traceMethod(handler, {
+                spanName,
+                attributes,
+              })(data);
+            }
+            return handler(data);
```

Comment:
add else for clarity

```suggestion
            } else {
              return handler(data);
            }
```